File: estimators/est_quadraticdirectbg1D.m
function xest=est_quadraticdirectbg1D(photonsi,L)
if length(photonsi)>3
    i1=3;i2=1;i0=5;
else
    i1=1;i2=2;i0=3;
end
p=photonsi([i1 i2 i0]);%for testing:
xest=(-(L*p(1)) + L*p(2))/(4*(2*p(3) - p(1) - p(2)));
xest(abs(xest)>L)=NaN;
end

File: python/estimators/est_quadraticdirectbg1D.py
import numpy as np

def est_quadraticdirectbg1D(photonsi, L):
    if len(photonsi) > 4:
        i1, i2, i0 = 2, 0, 4
    elif len(photonsi) == 3:
        i1, i2, i0 = 0, 1, 2
    elif len(photonsi) == 2:
        return q2point(photonsi, L)
    
    p = photonsi[np.array([i1, i2, i0])]  # for testing
    xest = (-(L*p[0]) + L*p[1])/(4*(2*p[2] - p[0] - p[1]))

    relmax=0.75
    xest[np.abs(xest)>(relmax*L)] = relmax*L  # extrapolation does not work well

    return xest

def q2point(photons, L):
    raise NotImplementedError("I don't know how to handle two points.")